torch/backends/cudnn/__init__.py

```diff
@@ -155,7 +166,7 @@ def set_flags(_enabled, _benchmark, _deterministic, _verbose):
     return orig_flags
 
 @contextmanager
-def flags(enabled=False, benchmark=False, deterministic=False, verbose=False):
+def flags(enabled=False, benchmark=(False if cuda is not None else True), deterministic=False, verbose=False):
```

Comment:
Let's use the same default value for both cudnn and miopen?

Reply:
Actually, I made this change deliberately, because the MIOpen path currently only supports `benchmark=True` mode. If we keep the default value `False` here, would that not contradict the warning printed from Module.cpp above?

Reply:
IIIC, here if we keep using False as the default value, ROCm users who have not provided an explicit benchmark value will get a warning (once) but their code still run fine?

Reply:
I restored the default value for `benchmark` parameter to `False`. Below is the behavior with this change:
```
>>> import torch
>>> import torch.backends.cudnn
>>> torch.backends.cudnn.benchmark
False
>>> torch.backends.cudnn.benchmark = True
>>> torch.backends.cudnn.benchmark = False
Warning: Disabling benchmark mode for MIOpen is NOT supported. Overriding value to True (operator() at /data/pytorch_jithunnair_fork/torch/csrc/Module.cpp:455)
>>> exit()
```
In other words, importing the cudnn module wouldn't throw any warnings, it's only when someone would try to set it to False that the warning would pop up